### PR TITLE
FIX: Division by 0 in `scil_dti_metrics`

### LINKED ISSUE #884: Division by 0
In the pytest of scil_dti_metrics:

RuntimeWarning: divide by zero encountered in log
    D[..., 6] = -np.log(b0)

This originates from dipy/reconst/dti.py:2120, which is in method lower_triangular, which is called through our own line 215:     tenfit = tenmodel.fit(data, mask)

### PR DESCRIPTION
**Pull request by AntoineTheb**
# Quick description

Closes #884. 

## Type of change

Check the relevant options.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Provide data, screenshots, command line to test (if relevant)

See tests

# Checklist

- [X] My code follows the style guidelines of this project (run [autopep8](https://pypi.org/project/autopep8/))
- [ ] I added relevant citations to scripts, modules and functions docstrings and descriptions
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I moved all functions from the script file (except the argparser and main) to scilpy modules
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by karanphil — APPROVED** [review]
Looks good! I'd be curious to know why the residuals are computed slice by slice as @AntoineTheb said. Maybe @mdesco or @arnaudbore or @frheault   know?

**Comment by AntoineTheb** [comment]
> Looks good! I'd be curious to know why the residuals are computed slice by slice as @AntoineTheb said. Maybe @mdesco or @arnaudbore or @frheault know?

I'm fairly sure it is done to reduce memory consumption. We could spend a few hours preemtively optimizing it by chunking the volume and multiprocessing but I don't think this is necessary right now @arnaudbore .

**Review by arnaudbore — APPROVED** [review]
